fn/src/Web/Fn.hs

```diff
@@ -423,6 +423,10 @@ instance FromParam Text where
   fromParam [x] = Right x
   fromParam [] = Left ParamMissing
   fromParam _ = Left ParamTooMany
+instance {-# INCOHERENT #-} FromParam String where
```

Comment:
Can you tell me a little more about why "INCOHERENT" is needed here? I've never seen it before.

Reply:
Hi. The reason is that `String` is actually `[Char]` and so GHC complains that it is overlapped with the `FromParam [a]` instance below. (According to the [doc](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#instance-resolution), GHC will not take context (i.e. constrains before `=>`) into account when matching instances.)
One solution is to add "INCOHERENT" pragma according to [this](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#instance-overlap). I am not sure if there is a better solution.

Reply:
That makes sense. Let me think a little about whether this feature is worth that extra bit of type complexity.  Regardless I appreciate you taking the time to make this PR

Reply:
No problem. Thank you very much!